### PR TITLE
Add manifest map test

### PR DESCRIPTION
**Pull request by SeaDve**

### CODE REVIEW

File: src/test/suite/extension.test.ts [hunk 1]
```diff
@@ -31,6 +34,74 @@ suite('command', () => {
...
+        assert.equal(map.get(manifestBUri), undefined)
+
+        assert(map.isEmpty())
+        assert.equal(map.getFirstItem(), undefined)
+    })
+
+    test('forEach', function () {
+        const map = new ManifestMap()
+
+        map.forEach(() => assert(false))
```
Comment (bilelmoussaoui) [inline src/test/suite/extension.test.ts:86]: What is the point of this assertion?

Comment (SeaDve) [reply src/test/suite/extension.test.ts:86]: that forEach would not never call the callback

Comment (SeaDve) [reply src/test/suite/extension.test.ts:86]: there's probably more explicit way to fail a test

File: src/manifestMap.ts [hunk 2]
```diff
@@ -31,7 +31,11 @@ export class ManifestMap {
     }
 
     getFirstItem(): Manifest | undefined {
-        return Array.from(this.inner.values())[0]
+        for (const [, manifest] of this.inner) {
+            return manifest
+        }
+
+        return undefined
```
Comment (bilelmoussaoui) [inline src/manifestMap.ts:38]: I am not sure about this change, what are the benefits of  going through a for if we can get the first item from the map?

Comment (SeaDve) [reply src/manifestMap.ts:38]: So constructing an array won't be needed?

Comment (bilelmoussaoui) [reply src/manifestMap.ts:38]: You can do that with `const key = this.inner.keys().next().value; return this.inner[key]` no?

Comment (SeaDve) [reply src/manifestMap.ts:38]: I did that before, but we need to cast it to string for eslint to not complain

Comment (SeaDve) [reply src/manifestMap.ts:38]:
```ts
const key = this.inner.keys().next().value as string | undefined

if (key !== undefined) {
     return this.inner.get(key)
}

return undefined
```

The for loop seems more elegant for me

Comment (SeaDve) [reply src/manifestMap.ts:38]: image

it's complaining here :shrug:

Comment (bilelmoussaoui) [reply src/manifestMap.ts:38]: I would drop this commit for now then, i will give it a try

**Comment by bilelmoussaoui** [comment]
Thanks!

**Pull request merged**